shelley/chain-and-ledger/formal-spec/properties.tex

```diff
@@ -1,10 +1,312 @@
+      & \text{by Lemma~\ref{lemma:value-sum-pres-2}}
+      \\
+      & \Val{(\var{rewards}\setminus\var{wdrls})} + w
+      & \text{by definition}
+      \\
+      & \Val\left(\var{rewards}\unionoverrideRight\{(w, 0) \mid w \in \dom \var{wdrls}\}\right) + w
+    \end{array}
+  \end{equation*}
+  Therefore $\Val(s) = \Val(s')$ in the base case.
+
+  In the inductive case, we look at the inductive rule $\mathsf{Seq{-}delg{-}ind}$.
+  But lemma then follows directly from Lemma~\ref{lemma:deleg-pres-of-value}.
+\end{proof}
+
+\begin{lemma}
+  \label{lemma:poolreap-pres-of-value}
+  For all environments $e$, epoch $\epsilon$, and states $s$, $s'$, if
+  \begin{equation*}
+    e\vdash s\trans{\hyperref[fig:rules:pool-reap]{poolreap}}{\epsilon}s'
+  \end{equation*}
+  then
+  \begin{equation*}
+    \Val(s) = \Val(s')
+  \end{equation*}
+\end{lemma}
+
+\begin{proof}
+  The $\mathsf{POOLREAP}$ value is contained in $\var{deposits}$, $\var{treasury}$, and $\var{rewards}$.
+  Notice that $\var{unclaimed}$ is added to $\var{treasury}$ and subtracted from the $\var{deposits}$.
+  Moreover, $\var{refunded}$ is subtracted from $\var{deposits}$.
+  (We will prove that $\var{deposits}-\var{unclaimed}+\var{refunded}$ is non-negative in a subsequent theorem.)
+  It therefore suffices to show that
+  \begin{equation*}
+    \begin{array}{r@{~=~}l}
+    \Val(\var{rewards}\unionoverridePlus\var{refunds})
+    & \Val(\var{rewards}) + \Val(\var{refunds})
+    \\
+    & \Val(\var{rewards}) + \var{refunded}
+    \end{array}
+  \end{equation*}
+  But this is clear from the definition of $\unionoverridePlus$.
+\end{proof}
+
+\begin{lemma}
+  \label{lemma:ru-pres-of-value}
+  For all mappings $b$ of blocks made, epochs $\epsilon$, and epoch states $s$,
+  \begin{equation*}
+    \Val(s) = \Val(\fun{applyRUpd}~(\fun{createRUpd}~b~s)~\epsilon~s)
+  \end{equation*}
+\end{lemma}
+
+\begin{proof}
+  In the definition of $\fun{applyRUpd}$ in Figure~\ref{fig:functions:reward-update-application},
+  we see that:
+  \begin{equation*}
+      \Delta t + \Delta r + \Val(rs) + \Val(\var{update_{rwd}}) + \Delta d + \Delta f = 0
+  \end{equation*}
+  In the definition of $\fun{createRUpd}$ in Figure~\ref{fig:functions:reward-update-creation},
+  we see that:
+  \begin{equation*}
+    \begin{array}{r@{~=~}l}
+      \var{rewardPot} & \var{feeSS} + \Delta r_1 \\
+      \var{R} & \var{rewardPot} - \Delta t_1 \\
+      \Delta t_2 & R \\
+    \end{array}
+  \end{equation*}
+  Therefore
+  \begin{equation*}
+    \begin{array}{c}
+      (\var{feeSS} + \Delta r_1) - \Delta t_1 - \Val(rs) = \Delta t_2 \\
+      0 = (\Delta t_1 + \Delta t_2 ) - \Delta r_1 + \Val(rs)- \var{feeSS} \\
+    \end{array}
+  \end{equation*}
+  So it suffices to show that $\Delta d + \Val(\var{update_{rwd}}) = \var{rewards}_{\var{mir}}$.
+  Notice that
+  \begin{equation*}
+    \begin{array}{r@{~=~}l}
+      \Val(\var{update}_{rwd}) & \Val(registered) + \Val(newlyRegistered')
```

Comment:
it's actually `newlyRegister'`  but feel free to change either here or in the `createRupd` function for consistency

Reply:
thanks!